Mathlib/Algebra/Ring/Int.lean

```diff
@@ -61,10 +62,13 @@ These also prevent non-computable instances like `Int.normedCommRing` being used
 these instances non-computably.
 -/
 
-instance instCommSemiring : CommSemiring ℤ := by infer_instance
-instance instSemiring     : Semiring ℤ     := by infer_instance
-instance instRingInt      : Ring ℤ         := by infer_instance
-instance instDistrib      : Distrib ℤ      := by infer_instance
+instance instCommSemiring : CommSemiring ℤ := inferInstance
+instance instSemiring     : Semiring ℤ     := inferInstance
+instance instRingInt      : Ring ℤ         := inferInstance
+instance instDistrib      : Distrib ℤ      := inferInstance
```

Comment:
This change means that any use of those instances comes at the cost of unfolding `inferInstance`. In general, please use `infer_instance` the tactic instead of `inferInstance` the term when defining instances

Reply:
`infer_instance`  the tactic is defined as `exact inferInstance`. So I do not see how this makes a noticeable difference. In fact, trying both versions for `Zero Nat` shows that the term uses the exact same number of heartbeats (and in other cases, the term seems to be even slightly faster).
I can revert this if you insist, though.

Reply:
Hmm, very strange. This was not the behaviour in Lean 3. Mind bringing it up on Zulip?

Reply:
Done [here](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/inferInstance.20vs.2E.20by.20infer_instance/near/432863769).

Reply:
According to Mario, which version we use does not really matter, so I'll leave it as it is now.
Is there anything else?

Reply:
No, that's it